### ISSUE TITLE
Snakes can open the doors to the Federal Reserve

### ISSUE THREAD

**Issue by Jawshy** [ArmaLife/Framework#744]
As the title says, snakes can open the doors to the Federal Reserve even when said doors are locked. This occurs because snakes are spawned near the client and open doors when they pass through them. This is an issue for self-explanatory reasons. 

##### Reproducible: 
- Yes, every time. 

Game version: [1.60.136266](https://forums.bistudio.com/topic/183855-release-candidate-branch-announcements/?p=3023745) RC
Mission version: v4.4r2 as of https://github.com/ArmaLife/Framework/commit/6e70557bf3ab15da4266cf5a827123a9a98b6008

**Comment by NiiRoZz** [comment]
For the 3. I think will need to move script inside server for creating bomb when finished with publicVariableServer and when variable it's true explode bomb; have issue because bomb created on client started Timer

**Comment by GetSomePanda** [comment]
No. 2 : https://github.com/GetSomePanda/Framework/commit/e0a2bfac23829f5ce0ea206e99636a9f1ad76dc5

**Comment by danielstuart14** [comment]
Do a pull request pls. @GetSomePanda

**Comment by GetSomePanda** [comment]
https://github.com/ArmaLife/Framework/pull/782

**Comment by Jawshy** [comment]
@GetSomePanda, thanks.

**Comment by GetSomePanda** [comment]
No. 3 https://github.com/ArmaLife/Framework/pull/786

**Comment by Jawshy** [comment]
2 down, 1 to go.

**Comment by GetSomePanda** [comment]
Arma 3 bug, not sure how to fix this issue, the only thing I can think of is removing the snakes from the game when they spawn. Unless someone has an idea.

**Comment by danielstuart14** [comment]
We can remove snakes near the dome. That's the only thing I can think

**Comment by tkcjesse** [comment]
Or just remove snakes all together. No one actually cares about snakes yes?

**Comment by setoy** [comment]
Aren't snakes even client-sided and thus provide no ability for player-interaction at all? get rid of 'em... And let's hope PETA doesn't know about this issue :p

**Comment by baua1310** [comment]
we could use something like this: 

`File: fn_disableSnakes.sqf
Author: ColinM
Description: Remove ambient snakes.
Credits: Lala14
*/

[] spawn {
 while {true} do {
  {
   if ((agent _x isKindOf "Snake_random_F")) then { deleteVehicle agent _x; };
  }forEach agents;
 sleep 2;
 };
};`

and replace `true` by a variable switched by a trigger around the federal reserve.

Source: [https://www.gtxgaming.co.uk/forum/forum/main-category/other-games/arma-3/arma-3-altis-life/65107-remove-snakes-increase-fps](https://www.gtxgaming.co.uk/forum/forum/main-category/other-games/arma-3/arma-3-altis-life/65107-remove-snakes-increase-fps)

**Comment by BoGuu** [comment]
I gave the code for this to Panda

**Comment by GetSomePanda** [comment]
```
_trigger = createTrigger ["EmptyDetector", POSITION OF DOOR, false]; 
_trigger setTriggerArea [6, 6, 0, false];
_trigger setTriggerActivation ["ANY", "PRESENT", true]; 
_trigger setTriggerStatements ["this", "{ if (agent _x distance thisTrigger < 6) then {deleteVehicle (agent _x);}; } forEach agents;", ""];
```

**Comment by torchgodz** [comment]
![screenshot2](https://cloud.githubusercontent.com/assets/5185094/16319634/fe28b096-3961-11e6-8e4f-f85071b20f49.jpg)
I keep getting this error when the bomb tries to go off and the civ cannot get the gold and the cops cannot fix the vault. Unsure why this is happening cause the variable is defined in multiple places.

**Comment by Jawshy** [comment]
@torchgodz, is the variable name of your vault `fed_bank` in the [mission.sqm](https://github.com/ArmaLife/Framework/blob/60639fc593b8f20e623c56ba3a5d1697e1e67987/Altis_Life.Altis/mission.sqm#L15924-L15966)?

**Comment by torchgodz** [comment]
It was named fed_bank_1 did not even think of that as a possibility thank you for the help.

**Comment by danielstuart14** [comment]
@BoGuu push the code

**Comment by NiiRoZz** [comment]
Only way to delete snakes, it's waituntil like this : 

```
waitUntil {
	{
		if(agent _x isKindOf "Snake_random_F" || agent _x isKindOf "Rabbit_F") then {deleteVehicle agent _x};
	} forEach agents;
	sleep 10;
	false;
};
```
